firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Util/COBS.h

```
@@ -49,25 +50,32 @@ namespace Pufferfish::Util {
 /// \sa http://www.jacquesf.com/2011/03/consistent-overhead-byte-stuffing
 
 /// \brief Encode a byte buffer with the COBS encoder.
-/// \param buffer A pointer to the unencoded buffer to encode.
+/// \param buffer A ByteVector to the unencoded buffer to encode.
 /// \param size  The number of bytes in the \p buffer.
-/// \param encodedBuffer The buffer for the encoded bytes.
+/// \param encodedBuffer The ByteVector for the encoded bytes.
 /// \returns The number of bytes written to the \p encodedBuffer.
 /// \warning The encodedBuffer must have at least getEncodedBufferSize()
 ///          allocated.
```

Since we're now passing in ByteVectors, we can do the size checks instead of assuming that the output buffer has at least a certain size allocated (or else we go out-of-bounds in memory), and we can return an error if this condition is not met, so we can remove this warning from both functions.